### ISSUE TITLE
Editing with Synchronised Clips Causes Random Glitches During Playback

### ISSUE THREAD

**Issue by tangierc** [latenitefilms/BRAWToolbox#115]
I am having some strange playback anomalies when I render a clip or collapse an adjacent clip I get strange frame jumps during playback and the footage becomes out of sync with already synced connected audio (this is not the audio from the BRAW toolbox). I'll post video samples to latenight films since it's a work in progress.

1) Here's sample images for the collapsing issue.
2)The other issue pertains to the next BRAW Toolbox clip (after the stacked RED clips) whereas, after being rendered for smoother playback, the clips were no longer in sync and also jumped in time.

Strangely I went to delete generated render files and that wasn't even an option.

FCP Playback was in proxy preferred mode
Machine specs: MacBook Pro M2 Max 64 GB RAM / macOS 13.3.1 / FCP 10.6.5
![Before Collapse Screenshot 2023-04-17 at 8 37 22 PM](https://user-images.githubusercontent.com/9577806/232667383-e5990ba4-c02c-4ec0-8b07-34c3ff82970d.png)
![After Collapse Screenshot 2023-04-17 at 8 37 52 PM](https://user-images.githubusercontent.com/9577806/232667389-9cae8edc-59bc-4bd2-959a-0a028f0b18f5.png)

**Comment by latenitefilms** [comment]
I'll have a look at the files you sent through soon. In the meantime, does making it into a Compound Clip fix it?

In the past we've had some other users report strange things like this when using Synchronised Clips, but when they switching to generating Multicam Clips from BRAW Toolbox, the problems went away - even though the Multicam Clip simply contains the same Synchronised Clip.

**Comment by tangierc** [comment]
Hello Chris. Turning the BRAW Toolbox synchronized clip to a compound clip does not fix the issue.  This is strange behavior; rendering and / or collapsing a clip on a BRTB clip to cause this. Do you have any thoughts on what would make things become out of sync like that or jump frames?  I copies that segment to a fresh timeline just to see if the timeline had any corruption issues and the results were the same.

Tangier

**Comment by latenitefilms** [comment]
It's a weird one - it has to be some kind of Final Cut Pro bug. I'll reach out to Apple again and see if they have any words of wisdom - but I'm guessing they just never really expected Synchronised Clips to contain just a Generator and an Audio Clip.

Again, I've only ever heard of this happening when people are using Synchronised Clips from BRAW Toolbox - it doesn't seem to happen when you use the **Create Multicam Clips** option in BRAW Toolbox - which again, is totally weird, as all this option does is wrap the Synchronised Clip within a Multicam Clip.

I would have assumed that wrapping the Synchronised Clip in a Compound Clip would have the same result - but apparently not.

Have you tried trashing Final Cut Pro's preferences?

What happens if you **Open Clip** on the Synchronised Clip, and then select both the Generator & Audio Clip and then create a Compound Clip? Does that fix it in the timeline?

**Comment by tangierc** [comment]
Chris, I did go into the BRAW clip and make turn the elements into a compound clip and that did fix the issue. I only tested it on one clip and it stayed in sync in the timeline with our better audio.  I then undid that and made the compound clip again and the BRAW clip in the timeline was all black and played nothing. Repeated the process one more time and it worked again. I did not test rendering the clip or collapsing an adjacent clip on this clip (now as a compound clip) internally; both of those operations being what causes this problem.

So it seems like this will work. However having to go through every clip on the timeline and making them compound clips or perhaps even the browser clips for safety before editing with them is practical. I hope this is isn’t the only option.


Thanks

**Comment by latenitefilms** [comment]
Thanks for checking. It definitely seems like some kind of weird Final Cut Pro bug rather than anything specifically BRAW Toolbox related. The workaround is basically to always create Multicam Clips from BRAW Toolbox when you start editing (or if you need to sync with Sync-N-Link, leave it as Synchronised Clips and Sync-N-Link will create Multicam's anyway so the result is the same).

Why it's happening is currently unclear. I'm still on Monterey - so it's entirely possible they inadvertently broke something in Ventura - as there seems to be a few weird regressions with Ventura and Audio Units, etc.

If you've already done a lot of editing/organising with the Synchronised Clips then that makes things a bit more tricky.

I'll fire up a Ventura machine and see if I can reproduce and come up with any workarounds.

**Comment by tangierc** [comment]
Thanks for looking into it. I sincerely appreciate the help. We’ve done a lot of work on this film already and are racing to a deadline for delivering to other departments.

Tangier

**Comment by latenitefilms** [comment]
Sorry Tangier, but I'm really struggling to reproduce this issue. Any chance you could send me a library with a single BRAW clip and RED file so that I can try and reproduce?

Ideally something just like this:

<img width="402" alt="image" src="https://user-images.githubusercontent.com/22286696/233902883-5fced4b7-9e04-4fee-93a9-ee1df80b23dd.png">

What frame rate is the Project/Timeline? What frame rate is the BRAW? What frame rate is the RED?

I'm still thinking it must be some very random/bizarre Final Cut Pro bug/glitch - because if the Synchronised Clip is playing fine in the browser, and if it's playing time in the timeline until you collapse, then there must be something going wrong very deep within Final Cut Pro.

**Comment by latenitefilms** [comment]
Also, if you turn on **Window > Source Timecode** is the timecode correct? Does the video timecode match the audio timecode for the BRAW file?

**Comment by tangierc** [comment]
Chris I am uploading a package to you now. I did a full timeline render and a lot majority of my BRTB clips became out of sync and the ones that did play fine after rendering couldn't further be edited on the heads or tails without going out of sync or jumping. Rendering is not an option for me. I can only verify if my timeline is good by exporting the entire project or rendering small portions at a time and fixing issues progressively until I get to the end and then export.

Oddly, I don't see where the render files are going.  I have a folder designated for my FCP project media, but see no render files in there.

**Comment by latenitefilms** [comment]
I will have a proper look at this when I get into the office this morning.

Basically, as previously discussed, this looks like it's just a Final Cut Pro bug with Synchronised Clips. It passes the wrong timecode values to effects in some (but weirdly, not all) cases. AFAIK the only workaround is to wrap the synchronised clip in a Multicam or Compound Clip.

As an experiment, can you please try:

1. Using the Toolbox to send a FCPXML to Resolve. My GUESS is that everything will be correct in Resolve.

2. Try just wrapping your entire timeline in a Compound Clip and re-exporting. For example, duplicate your timeline/project, select all, and then create a Compound Clip. This MIGHT allow you to render your timeline correctly.

In the next BRAW Toolbox update I'll make "Create Multicam Clips" the default and update the documentation to explain.

In terms of fixing your current job, it's tricky, because every time you "cut" a synchronised clip in the timeline, you now have two unique instances of the synchronised clip - which is why you can rename them differently, and when you "Clip > Open Clip" on them, they can have different insides. They also have no direct relation to their browser parents - i.e. unlike Multicam Clips, if you "Open Clip" on a synchronised clip in the Browser, it'll have no impact on the clips in the timeline. This means, it's not quite as simple as replacing all your synchronised clips with Multicam Clips as they inherently behave differently.

I'll do some more digging today.

**Comment by tangierc** [comment]
Thanks Chris for checking in out. The Writer/Director and I are baffled why clips that are in sync with the audio become out of sync after rendering, collapsing over another clip and can’t be touched to edit further after rendering without losing sync or weird frame jumps.  We’re spot checking the entire cut and have just relegated to a workflow that whereas we cannot render any BRTB clips or the timeline at all reliably.  The things is is that it’s not consistent. Some clips we can do all of the above operations and things are fine, but many are not. The one thing that is consistent is that once a clip is rendered for better playback, I can’t touch it in any way if I want it to stay in sync with the better recorded audio and without the frame jumps.

I wonder if Apple is even aware of this if it is a bug.

Thanks again.

Tangier

**Comment by latenitefilms** [comment]
@tangierc - Just to confirm... is that "Test 4 Chris" timeline/project not working on your machine? I just opened up the Final Cut Pro library and manually relinked the files, and it's all playing in sync for me.

HOWEVER, after messing with the timeline a bit to try and break it, I found something really interesting.

Here's the manipulated timeline:

<img width="1162" alt="image" src="https://user-images.githubusercontent.com/22286696/236960566-3a748c4d-5dff-4334-8beb-5a5e3f7c4961.png">

If I holding down the right arrow key, and watch it play in near-real-time, everything looks and plays correctly. But if I press play, then I'm seeing the same jumping frames/weirdness you're seeing.

I think this basically confirms it's not BRAW Toolbox that's doing something funky (because it plays the correct frames when going frame by frame) - but something is going funky during Final Cut Pro's playback.

Thanks so much for supplying these files - HUGELY appreciated! I now have something I can demonstrate to the Apple engineers, and hopefully they can provide a bug fix or workaround.

**Comment by latenitefilms** [comment]
It's also worth noting that if I double clicked each clip in the timeline and put the Generator inside a Compound Clip - it fixed things back on the timeline.

<img width="1486" alt="image" src="https://user-images.githubusercontent.com/22286696/236966987-0ef7acdd-64d2-4c8a-b419-b3881ba4c52c.png">

**Comment by latenitefilms** [comment]
I'm going to close this issue in favour of #128 - however, feel free to keep replying here if needed.

**Comment by tangierc** [comment]
Hello Chris. It’s my main timeline that doesn’t work and a previous edit of that timeline.  These clips were copied from that timeline into a new timeline for you.


Working on another issue (Final Cut Pro error -536870181 <https://discussions.apple.com/thread/254849318?answerId=259034670022#259034670022&cid=em-com-apple_watches_email_thread_owner-message-en-us-11192020#259034670022#259034670022>) I posted on Apple support communities, user joema discovered that I my computer didn’t have the Apple Pro Video Formats installed. That I find odd and additionally software update doesn’t make it available as it should. I installed it manually. 

I wanted to mention this is case not having it installed when I posted this has any impact or relation to the problem. Looking at the formats list on https://support.apple.com/kb/DL2050?locale=en_US, it doesn’t seem like it should in this case, but I want to make sure you have the full picture of my computer and pro apps setup just in case there was anything meaningful and/or useful that I hadn’t considered.

Tangier 

On May 9, 2023, at 1:38 AM, Chris Hocking ***@***.***> wrote:


@tangierc <https://github.com/tangierc> - Just to confirm... is that "Test 4 Chris" timeline/project not working on your machine? I just opened up the Final Cut Pro library and manually relinked the files, and it's all playing in sync for me.

HOWEVER, after messing with the timeline a bit to try and break it, I found something really interesting.

Here's the manipulated timeline:

 <https://user-images.githubusercontent.com/22286696/236960566-3a748c4d-5dff-4334-8beb-5a5e3f7c4961.png>
If I holding down the right arrow key, and watch it play in near-real-time, everything looks and plays correctly. But if I press play, then I'm seeing the same jumping frames/weirdness you're seeing.

I think this basically confirms it's not BRAW Toolbox that's doing something funky (because it plays the correct frames when going frame by frame) - but something is going funky during Final Cut Pro's playback.

Thanks so much for supplying these files - HUGELY appreciated! I now have something I can demonstrate to the Apple engineers, and hopefully they can provide a bug fix or workaround.

**Comment by latenitefilms** [comment]
I don't think that the [Pro Video Formats 2.2.4](https://support.apple.com/kb/DL2050?locale=en_AU) would have any direct impact on BRAW Toolbox. FWIW - the Pro Video Formats update should trigger when you install Final Cut Pro, but it comes through System Updates rather than the App Store, so if you have automatic updates disabled, you'd probably never notice.

I've sent all the information I can to Apple (Feedback Assistant ID: FB12179665), so fingers and toes crossed they look into it, and if it's a bug in BRAW Toolbox, they offer some insight, but I think, based on all of the above, it's probably a bug/oversight in Final Cut Pro that they'll need to fix on their end - and I don't see a fix happening quickly, especially as Final Cut Pro 10.6.6 is already most likely locked and ready for delivery.

Basically, I think the only good solution to your current film is to manually open each individual BRAW clip and then press OPTION+G (New Compound Clip) on the Generator. This should fix the issue, but it's obviously HUGELY painful and annoying to do it on a feature film! Assuming you're sending the FCPXML to DaVinci Resolve or elsewhere for grade/online, you probably also want to test for any issues with that workflow too.

I'm still trying to think of a way of automating this process, but it's just a bit complicated and tricky.

**Comment by tangierc** [comment]
Thanks for sending this to Apple Chris as well as a potential solution. I don’t think I can afford to go into each clip manually and do that. There are have been so many projects I’ve worked on where I’ve wished I had some sort of recursive procedure app to properly swap out clips with a similar counterpart or perform the task like your talking about. I’ve even reached out to Gregory at Intelligent Assistance. I would also fear attempting this and having something else show up that I have to manually undo.

For now my only option it seems is to correct the problems I and the writer/director have found and never render, and only rely on exporting the Quicktime to check.

We will be sending to resolve for color.

Thanks Chris.

**Comment by latenitefilms** [comment]
Have you already done a test sending your timeline/project to Resolve? Even though the frames might appear wrong during playback/rendering/export in Final Cut Pro (but still look correct when going frame-by-frame and skimming), I believe this is just a Final Cut Pro player bug - so the FCPXML export to Resolve should still be correct.

Are you able to send me a FCPXML of your current project/timeline? I'll see if I can put something together that replaces the Synchronised Clips with Multicam Clips in the timeline.

**Comment by tangierc** [comment]
I just sent the XML Chris.  I believe the company doing our color is going to take my locked picture Quicktime, an XML and EDL and conform the edits to do color this time rather than have all of the media to reference. I know they’re using Da Vinci Resolve for sure. I’ll be processing the EDL using EDL-X.

I’ve tested sending an XML already and an EDL which at one point you helped with. I didn’t have these particular problems with sync, rendering, and pulling heads/tails after rendering at the time however.

Thanks,

Tangier

**Comment by latenitefilms** [comment]
Legend, thank you! Received. I'll have a proper look at this later today and tomorrow.
